src/v/pandaproxy/kafka_client_cache.cc

```diff
@@ -153,16 +156,17 @@ ss::future<> remove_client_if(List& list, Pred pred) {
         }
     }
     for (auto& item : remove) {
-        co_await item.client->stop()
-          .handle_exception([&item](std::exception_ptr ex) {
-              // The stop failed
-              vlog(
-                plog.debug,
-                "Stale client {} stop already happened {}",
-                item.key,
-                ex);
-          })
-          .finally([client{item.client}]() {});
+        co_await ss::with_semaphore(*item.client_sm, 1, [&item]() {
```

Comment:
Nice, I think this works.

Reply:
Yes, I was experimenting with different sem counters. I'll change to a mutex